client/my-sites/marketing/traffic/index.js

```diff
@@ -105,10 +106,12 @@ const SiteSettingsTraffic = ( {
 );
 
 const connectComponent = connect( ( state ) => {
+	const site = getSelectedSite( state );
 	const siteId = getSelectedSiteId( state );
 	const isAdmin = canCurrentUser( state, siteId, 'manage_options' );
 	const isJetpack = isJetpackSite( state, siteId );
-	const isJetpackAdmin = isJetpack && isAdmin;
+	const isEligible = isEligibleForSEOFeatures( site, state, siteId );
+	const isJetpackAdmin = isJetpack && isAdmin && isEligible;
```

Comment:
`isJetpackAdmin` seems used for other features than SEO like Stats or the WP.me Shortlinks. Why `isEligibleForSEOFeatures` is controlling these features?

Reply:
The issue I'm trying to solve here is that `isJetpackAdmin` returns `true` for a free / no plan Atomic site. There's no particular feature we're checking for, just Jetpack. But this check is used to determine whether to show the Stats and WP.me Shortlinks features. I guess it should really be checking for Analytics feature availability. Would having a `isEligibleForAnalyticsFeatures` helper or using `hasSiteAnalyticsFeature()` help clarify that?

Reply:
> The issue I'm trying to solve here is that `isJetpackAdmin` returns `true` for a free / no plan Atomic site.

Why is this an issue? AFAIK both Stats or the WP.me Shortlinks are features available for free sites, aren't they?

Reply:
Stats and WP.me Shortlinks are available to all Jetpack plans (including free) and Atomic sites. But they’re not available for Simple free sites, under Tools > Marketing > Traffic:

Simple | Atomic
------- | -------
<img width="1051" alt="Screen Shot 2021-06-07 at 2 10 31 PM" src="https://user-images.githubusercontent.com/1689238/121068476-529f6180-c79a-11eb-936a-d33784d17356.png"> | <img width="1055" alt="Screen Shot 2021-06-07 at 2 10 56 PM" src="https://user-images.githubusercontent.com/1689238/121068466-503d0780-c79a-11eb-8dc3-fcb6a84733b2.png">

We're trying to provide the same experience for free Atomic sites as the one we provide for Simple free. These settings weren't mentioned in the issue, but they're also Marketing -> Traffic settings that shouldn't be available / visible on Atomic free. I think there might be additional issues, like whether we show the ad removal upgrade banner and Cloudflare.

Reply:
Gotcha. As per the discussion in pd2C1Z-10-p2, I think it makes sense to remove/hide these features from Atomic sites on any plan (free or paid) in order to achieve parity across Simple and Atomic sites (although I recommend allowing users to turn on the features if they were deactivated in the past).

As per the helper function used for determining if a site is eligible for a feature, I don't see a clear correlation between WP.me shortlinks or Stats, so I like your suggestion of creating a more abstract `hasSiteAnalyticsFeature` function which checks if a given feature is available for a given site (e.g. `hasSiteAnalyticsFeature( state, siteId, 'shortlinks' )`).

Reply:
@mmtr I've removed this change from this PR in favor of investigating removing these from Atomic further. Follow-up issues:
- https://github.com/Automattic/wp-calypso/issues/53507
- https://github.com/Automattic/wp-calypso/issues/53508